cmake/OpenCVFindTengine.cmake

```diff
@@ -0,0 +1,142 @@
+# COPYRIGHT
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# License); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# AS IS BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Copyright (c) 2020, OPEN AI LAB
+# Author: qtang@openailab.com or https://github.com/BUG1989
+#
+
+# Default tengine source store directory .
+SET(DEFAULT_OPENCV_TENGINE_SOURCE_PATH ${OpenCV_BINARY_DIR}/3rdparty/libtengine/Tengine-1.12.0)
+
+IF( WITH_CUDA OR WITH_OPENCL OR WITH_CUDNN OR WITH_VULKAN OR X86 OR X86_64)
+	MESSAGE(STATUS "TENGINE:--  Not support . Turning Tengine_FOUND off.")
+
+	SET(Tengine_FOUND OFF)
+ELSE()
+	IF(OPENCV_TENGINE_INSTALL_PATH)
+
+		MESSAGE(STATUS "TENGINE:--  Set tengine lib dir by user ")
+
+		SET(Tengine_FOUND ON)
+		set(BUILD_TENGINE OFF)
+
+		SET(Tengine_INCLUDE_DIR   	${OPENCV_TENGINE_INSTALL_PATH}/include)
+		SET(Tengine_LIB 	${OPENCV_TENGINE_INSTALL_PATH}/lib/libtengine.a)
+
+	ELSEIF(EXISTS ${DEFAULT_OPENCV_TENGINE_SOURCE_PATH})
```

Comment:
Code below this line (downloading and compiling) should be moved to `3rdparty` directory.

Reply:
done